scripts/next-dev-cycle.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+set -eo pipefail
+
+# Default values
+DEV_MODE=${DEV_MODE:=false}
+MAIN_BRANCH=${MAIN_BRANCH:=main}
+NEXT_VERSION=""
+
+Help()
+{
+   # Display Help
+   echo "This tool prepares the project to the next development cycle."
+   echo ""
+   echo "Syntax: next-dev-cycle [OPTIONS]"
+   echo "options:"
+   echo "h    Display this guide."
+   echo ""
+}
+
+while getopts "hn::s" option; do
+   case $option in
+      h) Help
+         exit;;
+      *) ;;
+   esac
+done
+
+if [ "$DEV_MODE" != "false" ]; then
+  echo "WARNING: Dev mode enabled..."
+fi
+
+# Get current git branch
+CURRENT_BRANCH=$(git symbolic-ref --short HEAD)
+if [ "$CURRENT_BRANCH" != "$MAIN_BRANCH" ]; then
+    echo "ERROR: Run the release process from $MAIN_BRANCH."
+    exit 1
+fi
+
+echo "Preparing for next development cycle $CURRENT_BRANCH"
+
+# Compute release version if not provided
+if [ -z "$NEXT_VERSION" ]; then
+    NEXT_VERSION=$(poetry version patch -s --dry-run)
+fi
+
+# Extract the major and minor version components
+MAJOR_MINOR_VERSION=$(echo "$NEXT_VERSION" | awk -F. '{print $1 "." $2}')
+# Compute stable branch from new version
+STABLE_BRANCH="$MAJOR_MINOR_VERSION.x"
```

Comment:
Since you never use `MAJOR_MINOR_VERSION` again, and since you seem comfortable with `sed`, I would use that instead of `awk` (also, I recommend "Here Strings" over piping `echo` output):
```bash
STABLE_BRANCH=$(sed -E -e 's/([^.]+\.[^.]+).*/\1.x/' <<< $NEXT_VERSION)
```
or just do it directly in the shell with something like
```bash
STABLE_BRANCH=${NEXT_VERSION%.*}.x
```

Ditto for `scripts/update-version.sh`.

Reply:
This looks really good
```
STABLE_BRANCH=$(sed -E -e 's/([^.]+\.[^.]+).*/\1.x/' <<< $NEXT_VERSION)
```